### DISCUSSION TITLE
Geyser support

### DISCUSSION THREAD

**Discussion by vuryy** [mizarc/bell-claims#83] [Ideas]
This is one of the most awesome SMP friendly claim plugin i have seen. Only sad part is that it doesn't work with geyser(bedrock) clients. Their client crashes when anvil GUI opens and they kicked for "Exceeding packet rate limit"
If you could support or just fix the anvil part it would be a grand.

**Comment by mizarc** [comment]
Geyser support would be fantastic, but I'd have to do a lot of research to figure out how best to support it as the inventory interaction system seems to be inherently incompatible with it. I'll definitely try supporting it as a future consideration as the logistics behind it may be complicated.

**Comment by mizarc** [comment]
I'll open an issue page on this in the meantime, will see what I can do as support between versions on the inventory framework is already quite iffy even within minor Java edition versions.

**Comment by vuryy** [comment]
I did quick research as well and seems that anvil/smithing table GUIs are totally different in bedrock edition. I guess that is the reason why most plugin developers redirect players to chat box input. However seeing amazing work you have done so far, I doubt that method aligns with your vision

**Comment by Asylum-Media** [comment]
I'm looking into this myself. Are you ok with me trying to find a workaround @mizarc ?
I have some ideas on how to bring this to fruition.

**Reply by mizarc** [reply]
Feel free to find a solution. I don't have any immediate plans to support Geyser, but I'm happy to accept contributions if you can find an intuitive way around it.
